### ISSUE TITLE
Lambda Layer always in Status Change since layer refactoring

### ISSUE THREAD

**Issue by StefanV85** [aws-amplify/amplify-cli#7569]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

14.15.5

### Amplify CLI Version

5.0.1

### What operating system are you using?

Windows

### Amplify Categories

auth, function, api, hosting

### Amplify Commands

push, status

### Describe the bug

Hello,

My existing Amplify Project use Lambda Layers. I updated now to Amplify Cli Version 5.0.1. Since Amplify 5 there is a complete refactoring of the Layer Handling.

The Migration of my Lambda Layer was done. and i updated all lambda settings to always use the latest lambda layer.
it works fine. if i change my layer code, all lambda functions are updated to the newest version.

The Problem is, that after the Amplify Push, the layer has again (without coding change) the amplify status "change".
so if i would do an "amplify push" => everything works.
But if i would do an "amplify push" or "amplify status" directly after, the status of the layer is imediatly "Update".

The result is, that each amplify push, will result in an new amplfy layer version, even i doesn't change the layer code.
if the layer always have the status "change".

i tried to setup a new "hello world" amplify project, with a lambda function, and an lambda layer. The problem doesn't exist here. after the "amplify push", the status of the lambda layer is in Status "No Change".

Are there any tricks, how to fix my layer in the existing project? Or how can i find out the reason, why amplify set the status to "Update"?

Kind Regards
Stefan

### Expected behavior

After the Amplify Push, the layer has the status "No Change"

### Reproduction steps

1. amplify push => successfully "√ All resources are updated in the cloud"
2. amplify status => Layer has status "Change"
3. amplify push => New Layer Version is created. 

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by attilah** [comment]
@StefanV85 this is an issue that we try to replicate on our end with no success yet. Could you give more information about your layer and the project?

1. what runtime is this layer using?
2. what version of the CLI was the layer created with?
3. did you do the migration with 5.0.0 or 5.0.1?
4. is the layer referenced by any function?
5. can you send a file list without (node_modules or lib/python3.x directory) of the affected layer?

**Comment by StefanV85** [comment]
Hi @attilah 

I tried to narrow down the problem for a while today.
In generell i was able to reproduce the problem in a new amplify project. I just copied all my source code files (which are in the opt folder of the layer) in the new project. And The Problem is in the new amplify project again.

so it's defenetly not a problem with the "migration" of the layer refactoring.
Unfortunately i can't share my sourcecode here, because it's production code.

i wasn't able to narrow down the problem to specific files yet.

regarding your questions: 

1. Layer Runtime: NodeJS
2. Which Version created the lambda layer: because it happens in a new project too, we can use the version 5.0.1.
3. Migration happend with 5.0.1. But it's not related to the migration, because it also happens in new projects.
4. Yes i have Lambda Functions wihich use this layer. Setting: "Always  use the newest version"
5. Is the filelist still necassary? Because i see you already linked an pull request?  :-)
